app/assets/javascripts/arclight/context_navigation.js

```diff
@@ -308,9 +316,193 @@ class ContextNavigation {
   }
 }
 
+class CollectionSiblingTree {
+  setSiblingElements() {
+    this.$previousDocs = this.$currentDoc.prevUntil().slice(0, 2);
+    this.$nextDocs = this.$currentDoc.nextUntil().slice(0, 2);
+  }
+
+  updateRootElement() {
+    this.$currentDoc.addClass('al-hierarchy-highlight');
+    // Unlink the current component - just show the title
+    this.$currentDoc.find('.al-hierarchy-highlight a').contents().unwrap();
+  }
+
+  build() {
+    this.$elements = $('<div>');
+
+    if (this.$previousDocs.length > 0) {
+      if (this.$nextDocs.length > 0) {
+        this.$elements.append(this.$previousDocs.first());
+        this.$elements.append(this.$currentDoc);
+        this.$elements.append(this.$nextDocs.first());
+      } else {
+        this.$elements.append(this.$previousDocs.get().reverse());
+        this.$elements.append(this.$currentDoc);
+      }
+    } else {
+      this.$elements.append(this.$currentDoc);
+      this.$elements.append(this.$nextDocs);
+    }
+
+    this.$elements.find('.al-toggle-view-more').remove();
+    this.$elements.find('.collapse').remove();
+    this.$elements.find('hr').remove();
+  }
+
+  contructor(id, $doc) {
+    const headers = $doc.find(`article div[data-document-id="${id}"]`);
+    if (headers.length === 0) {
+      $.error(`Document is missing id=${id}`);
+    }
+    const firstHeader = headers[0];
+    // This is the root of the tree
+    this.$currentDoc = $(firstHeader.parentNode);
+    this.updateRootElement();
+    this.setSiblingElements();
+    this.build();
+  }
+}
+
+class ArcLightRequestAdapter {
+  /**
+   * Transmit the GET request to the server
+   * This is specific to the Blacklight/ArcLight CatalogController parameters
+   *   for faceting
+   */
+  getData(receivedCallback) {
```

Comment:
I really would prefer to an approach using `async`/`await` for returning the AJAX responses from this method, but I believe that this cannot be done without addressing Babel transpiling issues (please see https://github.com/projectblacklight/arclight/issues/869).